### PR TITLE
OkCoin websocket initial commit

### PR DESCRIPTION
**Pull request by jheusser**
Initial commit for websocket market data streaming. It's based on the reference implementation by OkCoin and already contains its trade functionality, but has not been wired up into the XChange interface yet. It's mostly reusing the existing market data DTOs.

`OkCoinSocketIODemo` shows the standard demo for streaming.

### CODE REVIEW

**Comment by jheusser** [comment]
Probably have to commit some code to make currency switching easier by default

File: xchange-okcoin/pom.xml [hunk 1]
```diff
@@ -29,6 +30,12 @@
 			<version>${project.version}</version>
 		</dependency>
 
+		<dependency>
+			<groupId>io.netty</groupId>
+			<artifactId>netty-all</artifactId>
+			<version>4.1.0.Beta3</version>
+		</dependency>
+
```
Comment (timmolter) [inline xchange-okcoin/pom.xml]: Can you put this under `<dependencyManagement>` in parent.pom?

Comment (jheusser) [inline xchange-okcoin/pom.xml]: thanks, will do

**Comment by timmolter** [comment]
Good work!!

**Pull request merged**